### PR TITLE
Create OIDC extension for  Flutter SDK 

### PR DESCRIPTION
**Pull request by Ariflo**
- Adds the OIDC Extension for Flutter SDK
- Updates Licensing to be Apache to match the rest of our SDKs
- Updates Demos and Versioning

### CODE REVIEW

File: packages/magic_sdk/lib/modules/base_module.dart [hunk 1]
```diff
@@ -29,4 +29,20 @@ class BaseModule {
     return provider.send(
         request: request, completer: Completer<JavaScriptMessage>());
   }
+
+  Future<JavaScriptMessage> sendToProviderWithList({
```
Comment (Ethella) [inline packages/magic_sdk/lib/modules/base_module.dart]: Curious, `sendToProvider` and `sendToProviderWithList` seem pretty identical. What's the use case for this new function 🤔

Comment (Ariflo) [reply packages/magic_sdk/lib/modules/base_module.dart]: I kept getting  the following error when I used `sendToProvider` as is:  🤔 

```
Unhandled Exception: {code: -32603, message: Internal error: object is not iterable (cannot read property Symbol(Symbol.iterator))}
```

Comment (Ariflo) [reply packages/magic_sdk/lib/modules/base_module.dart]: I see what you mean though, one sec, I'll try it again 😅

File: packages/magic_ext/oidc/lib/magic_ext_oidc.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  Future<String> loginWithOIDC(OpenIdConfiguration configuration) async {
+    var configJson = configuration.toJson();
+
+    JavaScriptMessage response = await sendToProvider(
+      method: OIDCMethod.magic_auth_login_with_oidc,
+      params: [configJson],
+    );
+
+    // Parse the JSON response
+    var decodedResponse = json.decode(response.message);
```
Comment (Ethella) [inline packages/magic_ext/oidc/lib/magic_ext_oidc.dart:23]: Oh this is more intuitive than my original implementation, feels like we can abstract this to a util function at a later time 👍

**Review by Ethella — APPROVED** [review]

**Pull request merged**